cns/restserver/ipam.go

```diff
@@ -118,6 +119,71 @@ func (service *HTTPRestService) GetPodIPConfigState() map[string]cns.IPConfigura
 	return service.PodIPConfigState
 }
 
+func (service *HTTPRestService) getIPAddressesHandler(w http.ResponseWriter, r *http.Request) {
+	var (
+		req           cns.GetIPAddressesRequest
+		resp          cns.GetIPAddressStateResponse
+		statusCode    int
+		returnMessage string
+		err           error
+	)
+
+	statusCode = UnexpectedError
+
+	defer func() {
+		if err != nil {
+			resp.Response.ReturnCode = statusCode
+			resp.Response.Message = returnMessage
+		}
+
+		err = service.Listener.Encode(w, &resp)
+		logger.Response(service.Name, resp, resp.Response.ReturnCode, ReturnCodeToString(resp.Response.ReturnCode), err)
+	}()
+
+	err = service.Listener.Decode(w, r, &req)
+	logger.Request(service.Name, &req, err)
+	if err != nil {
+		returnMessage = err.Error()
+		return
+	}
+
+	// Get all IPConfigs matching a state, and append to a slice of IPAddressState
+	resp.IPAddresses = make([]cns.IPAddressState, 0)
+	for _, podstate := range req.IPConfigStateFilter {
+		resp.IPAddresses = append(resp.IPAddresses, filterIPConfigsMatchingState(service.PodIPConfigState, podstate, func(ipconfig cns.IPConfigurationStatus, state string) bool {
+			return ipconfig.State == podstate
+		})...)
```

Comment:
No need to append. You can directly assign the result of filtering to `resp.IPAddresses`. You can also extract the anonymous function out to make it a bit easier to read.